dask/array/routines.py

```diff
@@ -1843,6 +1843,50 @@ def piecewise(x, condlist, funclist, *args, **kw):
     )
 
 
+def _select(*args, **kwargs):
+    split_at = len(args) // 2
+    condlist = args[:split_at]
+    choicelist = args[split_at:]
+    return np.select(condlist, choicelist, **kwargs)
+
+
+@derived_from(np)
+def select(condlist, choicelist, default=0):
+    # Making the same checks that np.select
+    # Check the size of condlist and choicelist are the same, or abort.
+    if len(condlist) != len(choicelist):
+        raise ValueError("list of cases must be same length as list of conditions")
+
+    if len(condlist) == 0:
+        raise ValueError("select with an empty condition list is not possible")
+
+    choicelist = [asarray(choice) for choice in choicelist]
+
+    try:
+        intermediate_dtype = result_type(*choicelist)
+    except TypeError as e:
+        msg = f"Choicelist elements do not have a common dtype: {e}"
+        raise TypeError(msg) from None
```

Comment:
I think this is the recommended style:

```suggestion
        msg = f"Choicelist elements do not have a common dtype."
        raise TypeError(msg) from e
```

Reply:
I like that, I just wanted to avoid changing numpy message.